### PR TITLE
Changes to output variables at 160 m and 320 m above ground

### LINKED ISSUE #671: Enable output of 160m and 320m wind from RRFS for renewable energy
Renewable energy users have asked for wind output at higher levels due to the construction of larger and larger wind turbines in recent years.

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
This PR enables output of TMP, UGRD, VGRD, PRES, and SPFH at 160 m and 320 m above ground level for renewable energy applications (for wind turbines of increasing height).  It also includes small changes related to the aerosol extinction due to smoke, dust, and climatological aerosol in RRFS.  

The new aerosol extinction variable to be read in, ext550, will be output when the realtime RRFS are upgraded to the authoritative ufs-weather-model.

The code was tested on Jet for the RRFS_CONUS_3km system, with no change in output except the addition of the new levels, and small changes in AOTK and VIS (due to the new extinction variable).

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Can you sync your branch with the UPP develop branch?

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Can you add change logs in the fortran code you modified?

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Can you add change logs in the fortran code you modified?

@WenMeng-NOAA Sorry about that!  I just added them.

File: sorc/ncep_post.fd/CLDRAD.f [hunk 1]
```diff
@@ -446,16 +447,9 @@ SUBROUTINE CLDRAD
...
       IF (IGET(735) > 0) THEN
-       IF (MODELNAME == 'RAPR') THEN
+       IF (MODELNAME == 'RAPR' .OR. MODELNAME == 'FV3R') THEN
          CALL CALPW(GRID1(ista:iend,jsta:jend),19)
          CALL BOUND(GRID1,D00,H99999)
-       ELSE IF (MODELNAME == 'FV3R') THEN
-         GRID1=SPVAL
-         DO J=JSTA,JEND
-           DO I=ISTA,IEND
-             if (AODTOT(I,J) < SPVAL) GRID1(I,J) = AODTOT(I,J)
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f:456]: @EricJames-NOAA You add changes for calculating column aod from the 3D field in UPP. Do you think the capability of reading aodtot from model output is needed to be kept in UPP?

Comment (EricJames-NOAA) [reply sorc/ncep_post.fd/CLDRAD.f:456]: @WenMeng-NOAA I'm OK with getting rid of this code; we can easily add it back in if needed.  Should I remove it?

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/CLDRAD.f:456]: @EricJames-NOAA That would be great if you can remove aodtot related code. Thanks!

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Do you have sample RRFS model history files including ext550 for my standalone testing?

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA Do you have sample RRFS model history files including ext550 for my standalone testing?

@WenMeng-NOAA I put some sample files here on Hera:

/scratch1/BMC/wrfruc/ejames/wen_ext550

These are 13km CONUS domain...I'm still working on getting a test case for the 3km CONUS domain.

**Comment by WenMeng-NOAA** [comment]
> > @EricJames-NOAA Do you have sample RRFS model history files including ext550 for my standalone testing?
> 
> @WenMeng-NOAA I put some sample files here on Hera:
> 
> /scratch1/BMC/wrfruc/ejames/wen_ext550
> 
> These are 13km CONUS domain...I'm still working on getting a test case for the 3km CONUS domain.

@EricJames-NOAA Could you copy them into Hera? My working computer has an issue to access Jet today. Thanks!

**Comment by EricJames-NOAA** [comment]
> > > @EricJames-NOAA Do you have sample RRFS model history files including ext550 for my standalone testing?
> > 
> > 
> > @WenMeng-NOAA I put some sample files here on Hera:
> > /scratch1/BMC/wrfruc/ejames/wen_ext550
> > These are 13km CONUS domain...I'm still working on getting a test case for the 3km CONUS domain.
> 
> @EricJames-NOAA Could you copy them into Hera? My working computer has an issue to access Jet today. Thanks!

Do you mean you need them on Jet?  I have some sample files here:

lfs4/BMC/wrfruc/ejames/RRFS/stmp/tmpnwprd/RRFS_conus_3km/2021051216/fcst_fv3lam

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Sorry, I thought the data location you provided from jet. Now I got it. Thanks!

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA I conducted UPP testing with model data from /scratch1/BMC/wrfruc/ejames/wen_ext550.
My test results are at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2021051216-after on Hera.
I saw the following expected changes:
```
1) additional 10 fields(T,U/V,P,SPFH) added in dataset PRSLEV.
2) VIS change as: 1237:59345473:VIS:surface:rpn_corr=0.993778:rpn_rms=2658.85
```
Please let me know if you any issues from my test. Thanks!

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA I conducted UPP testing with model data from /scratch1/BMC/wrfruc/ejames/wen_ext550. My test results are at /scratch1/NCEPDEV/stmp2/Wen.Meng/rrfs_2021051216-after on Hera. I saw the following expected changes:
> 
> ```
> 1) additional 10 fields(T,U/V,P,SPFH) added in dataset PRSLEV.
> 2) VIS change as: 1237:59345473:VIS:surface:rpn_corr=0.993778:rpn_rms=2658.85
> ```
> 
> Please let me know if you any issues from my test. Thanks!

@WenMeng-NOAA your results look good to me!  Thank you!

File: sorc/ncep_post.fd/CLDRAD.f (hunk 1 above)
Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/CLDRAD.f:456]: @EricJames-NOAA Would you like the AODTOT clean up in this PR or next PR?

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You might start the UPP RT tests on Hera and Orion. Thanks!

File: sorc/ncep_post.fd/CLDRAD.f (hunk 1 above)
Comment (EricJames-NOAA) [reply sorc/ncep_post.fd/CLDRAD.f:456]: @WenMeng-NOAA sorry for the delay.  I have the cleanup ready, I just need to commit the changes.  Stand by...

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 2]
```diff
@@ -31,8 +31,9 @@
 !> 2023-02-23 | Eric James    | Read coarse PM and aodtot from RRFS
 !> 2023-03-02 | Sam Trahan    | Read lightning threat index fields
 !> 2023-03-22 | WM Lewis      | Read RRFS effective radii (EFFRL, EFFRI, EFFRS)
-!> !> 2023-04-04 |Li(Kate Zhang)  |Add namelist optoin for CCPP-Chem(UFS-Chem) 
+!> 2023-04-04 |Li(Kate Zhang)  |Add namelist optoin for CCPP-Chem(UFS-Chem) 
 !         and 2D diag. output (d2d_chem) for GEFS-Aerosols and CCPP-Chem model.
+!> 2023-04-17 | Eric James    | Read in unified ext550 extinction for RRFS
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f]: @EricJames-NOAA Could you note that aodtot is removed?

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2. The file size of dataset PRSLEV in 3drtma test has a change but field by field value comparison is fine.  The baseline of 3drtma is needed to be recreated with this PR.

**Comment by FernandoAndrade-NOAA** [comment]
UPP RT tests have finished on Hera and Orion. All tests passed with the exception of the following changes in PRSLEV for 3drtma :
```
39:38622976:REFZR:entire atmosphere (considered as a single layer):rpn_corr=-nan:rpn_rms=undefined
40:38861307:REFZI:entire atmosphere (considered as a single layer):rpn_corr=-nan:rpn_rms=undefined
49:49740509:PBLREG:surface:rpn_corr=-nan:rpn_rms=undefined
703:635158661:REFZR:4000 m above ground:rpn_corr=-nan:rpn_rms=undefined
704:635396992:REFZI:4000 m above ground:rpn_corr=-nan:rpn_rms=undefined
706:636307658:REFZR:1000 m above ground:rpn_corr=-nan:rpn_rms=undefined
707:636545989:REFZI:1000 m above ground:rpn_corr=-nan:rpn_rms=undefined
763:706867267:POT:10 m above ground:rpn_corr=-nan:rpn_rms=undefined
764:707105598:SPFH:10 m above ground:rpn_corr=-nan:rpn_rms=undefined
```

**Comment by WenMeng-NOAA** [comment]
> UPP RT tests have finished on Hera and Orion. All tests passed with the exception of the following changes in PRSLEV for 3drtma :
> 
> ```
> 39:38622976:REFZR:entire atmosphere (considered as a single layer):rpn_corr=-nan:rpn_rms=undefined
> 40:38861307:REFZI:entire atmosphere (considered as a single layer):rpn_corr=-nan:rpn_rms=undefined
> 49:49740509:PBLREG:surface:rpn_corr=-nan:rpn_rms=undefined
> 703:635158661:REFZR:4000 m above ground:rpn_corr=-nan:rpn_rms=undefined
> 704:635396992:REFZI:4000 m above ground:rpn_corr=-nan:rpn_rms=undefined
> 706:636307658:REFZR:1000 m above ground:rpn_corr=-nan:rpn_rms=undefined
> 707:636545989:REFZI:1000 m above ground:rpn_corr=-nan:rpn_rms=undefined
> 763:706867267:POT:10 m above ground:rpn_corr=-nan:rpn_rms=undefined
> 764:707105598:SPFH:10 m above ground:rpn_corr=-nan:rpn_rms=undefined
> ```

@FernandoAndrade-NOAA Your tests on Hera and Orion are consistent with my test on WCOSS2. These discrepancies from rtma are from grib2 meta data changes. It would be fine for no change from field by field comparison.

**Comment by WenMeng-NOAA** [comment]
@EdwardColon-NOAA This PR would have impact on 3DRTMA since RRFS and 3DRTMA share the same UPP control file. Please let me if you have any suggestions/concerns.

**Comment by EdwardColon-NOAA** [comment]
Hi Wen,

The changes listed do not impact the 3D-RTMA development effort at the moment.  The PR should be fine from the 3D-RTMA perspective.

-Edward

**Comment by WenMeng-NOAA** [comment]
> Hi Wen,
> 
> The changes listed do not impact the 3D-RTMA development effort at the moment. The PR should be fine from the 3D-RTMA perspective.
> 
> -Edward

Thanks for confirming. @EdwardColon-NOAA

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Please approve this PR.

**Review by WenMeng-NOAA — APPROVED** [review]

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Pull request merged**